spec/system/project_spec.rb

```diff
@@ -180,18 +181,39 @@
         visit "/"
         click_on "New Project"
         expect(page.find("#data_sponsor").value).to eq sponsor_user.uid
+        fill_in "data_sponsor", with: sponsor_user.uid
         fill_in "data_manager", with: "xxx"
         fill_in "ro-user-uid-to-add", with: read_only.uid
         click_on "btn-add-ro-user"
         fill_in "rw-user-uid-to-add", with: read_write.uid
         click_on "btn-add-rw-user"
-        fill_in "directory", with: "test_project"
+        fill_in "directory", with: "test-project"
         fill_in "title", with: "My test project"
-        expect(page).to have_content("Project Directory: /td-test-001/")
         expect do
           click_on "Save"
-        end.not_to have_enqueued_job(ActionMailer::MailDeliveryJob).exactly(1).times
+        end.to have_enqueued_job(ActionMailer::MailDeliveryJob).exactly(0).times
         expect(page).to have_content("Invalid netid: xxx for role Data Manager")
+        expect(page).to have_content("New Project")
+      end
+    end
+    context "when the directory name has invalid characters" do
+      it "allows the user to create a project" do
+        sign_in sponsor_user
+        visit "/"
+        click_on "New Project"
+        fill_in "data_sponsor", with: sponsor_user.uid
+        fill_in "data_manager", with: data_manager.uid
+        fill_in "ro-user-uid-to-add", with: read_only.uid
+        click_on "btn-add-ro-user"
+        fill_in "rw-user-uid-to-add", with: read_write.uid
+        click_on "btn-add-rw-user"
+        fill_in "directory", with: "test?project"
```

Comment:
Could you check the validity value here? @bess put in an example here https://github.com/pulibrary/tiger-data-app/blob/main/spec/system/project_spec.rb#L175